packages/block-library/src/navigation/edit/index.js

```diff
@@ -231,23 +232,38 @@ function Navigation( {
 		clientId
 	);
 
+	const {
+		convert,
+		state: classicMenuConversionState,
+	} = useConvertClassicToBlockMenu( clientId );
+
+	const isConvertingClassicMenu =
+		classicMenuConversionState?.status === 'fetching';
+
 	// The standard HTML5 tag for the block wrapper.
 	const TagName = 'nav';
 
 	// "placeholder" shown if:
 	// - we don't have a ref attribute pointing to a Navigation Post.
+	// - we are not running a menu conversion process.
 	// - we don't have uncontrolled blocks.
 	// - (legacy) we have a Navigation Area without a ref attribute pointing to a Navigation Post.
 	const isPlaceholder =
-		! ref && ( ! hasUncontrolledInnerBlocks || isWithinUnassignedArea );
+		! ref &&
+		! isConvertingClassicMenu &&
+		( ! hasUncontrolledInnerBlocks || isWithinUnassignedArea );
 
 	const isEntityAvailable =
 		! isNavigationMenuMissing && isNavigationMenuResolved;
 
 	// "loading" state:
+	// - we are running the Classic Menu conversion process.
+	// OR
 	// - there is a ref attribute pointing to a Navigation Post
 	// - the Navigation Post isn't available (hasn't resolved) yet.
-	const isLoading = !! ( ref && ! isEntityAvailable );
+	const isLoading =
+		isConvertingClassicMenu ||
```

Comment:
Now `loading` UI state should show _immediately_ if we are running the conversion.